twake/frontend/src/app/scenes/Apps/Messages/Message/Parts/MessageContent.tsx

```diff
@@ -24,12 +23,16 @@ export default (props: Props) => {
   const [loadingAction, setLoadingAction] = useState(false);
   let loading_interaction_timeout: any = 0;
 
-  Collections.get('messages').useListener(useState, [
+  const listener = Collections.get('messages').useListener(useState, [
     props.message?.id,
     props.message?.front_id,
     'msgcontent',
   ]);
 
+  useEffect(() => () => {
+    Collections.get('messages').removeListener(listener);
+  }, []);
+
```

Comment:
Good spot but I think we can improve this and fix the memory leak in a single place:
in src/app/services/Depreciated/observable.js line 17 (useListener method)
You can put the useEffect here, useState should not be passed as a variable either (I know this is done on a lot of places in the code with old collections but it's wrong)

Reply:
Oh YES! Let's try this